### ISSUE TITLE
Access to XMLHttpRequest at ' ... ' from origin ' ... ' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.

### ISSUE THREAD

**Issue by mdutt247** [beyondcode/laravel-websockets#384]
![image](https://user-images.githubusercontent.com/59934065/82938648-c451a680-9faf-11ea-8349-991e68798f72.png)

The same error I solved on the local machine by adding `'useTLS' => false`, `'encrypted' => false` in `config/broadcasting.php` and `forceTLS:false`,  `'encrypted':false` in `resources/js/bootstrap.js`

When I try to run it on the live server, it gives CORS policy error
`Response to preflight request doesn't pass access control check`: `No 'Access-Control-Allow-Origin' header is present on the requested resource`.

I also tried downgrading pusher.js from 6.0.3 to 4.4 but got no success.

I have hosted this application on shared hosting on Laravel 5.8, and use Cloudflare.

**Comment by renwarkurd** [comment]
Same issue here

**Comment by kaikeSilva** [comment]
did you resolve this?

**Comment by renwarkurd** [comment]
No. I haven't.
I ended up not using it unfortunately.
Hope you guys find a solution for it.

**Comment by hassan-mo** [comment]
You can fix this URL issue:

Solution: 
In your post request make sure you have the full URL 'http://sockjs-mt1.pusher.com/pusher/app/....etc'

Error:
const siteUrl = 'http://sockjs-mt1.pusher.com/'
Post request `${siteUrl}/pusher/app/....etc`

**Comment by allaniftrue** [comment]
same here

**Comment by natnaelawel** [comment]
same issue here

**Comment by shaikhfoysal** [comment]
Anyone solve this issue.
I have the same problem.
` enabledTransports: ['ws', 'wss']` 
I did this but didn't solve the problem. What to do?

**Comment by worstestes** [comment]
> Anyone solve this issue.
> I have the same problem.
> ` enabledTransports: ['ws', 'wss']`
> I did this but didn't solve the problem. What to do?
> Local is good but not in the web.

Can you try setting `devServer.transportMode = "ws"` ?

**Comment by rennokki** [comment]
Duplicate of https://github.com/beyondcode/laravel-websockets/issues/403 (see you there)

**Comment by pushpendra2636** [comment]
@ECHO OFF
"C:\Program Files (x86)\Google\Chrome\Application\chrome.exe" --disable-web-security --disable-gpu --user-data-dir=~/chromeTemp
pause


/************
create a new file  and save file  .bat extension "like this :-  privateBrowser.bat" ,
and run the this file. and insert local path: http://localhost:4200
***********/

**Comment by Benudhar1996** [comment]
i am also getting this error

**Comment by rennokki** [comment]
Check https://github.com/beyondcode/laravel-websockets/issues/403

**Comment by ziagit** [comment]
I solved this problem by changing pusher-js version from 6.0.0 to 5.1.1. in (Laravel vuejs project)
